exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java

```diff
@@ -177,80 +179,55 @@ private Adapter() {}
    *
    * @param attributes the span attributes
    * @return a collection of Jaeger key values
-   * @see #toKeyValue(String, AttributeValue)
-   */
-  @VisibleForTesting
-  static Collection<Model.KeyValue> toKeyValues(Map<String, AttributeValue> attributes) {
-    List<Model.KeyValue> tags = new ArrayList<>(attributes.size());
-    for (Map.Entry<String, AttributeValue> entry : attributes.entrySet()) {
-      tags.add(toKeyValue(entry.getKey(), entry.getValue()));
-    }
-    return tags;
-  }
-
-  /**
-   * Converts a map of attributes into a collection of Jaeger's {@link Model.KeyValue}.
-   *
-   * @param attributes the span attributes
-   * @return a collection of Jaeger key values
-   * @see #toKeyValue(String, AttributeValue)
+   * @see #toKeyValue
    */
   @VisibleForTesting
   static Collection<Model.KeyValue> toKeyValues(ReadableAttributes attributes) {
     final List<Model.KeyValue> tags = new ArrayList<>(attributes.size());
     attributes.forEach(
         new AttributeConsumer() {
           @Override
-          public void consume(String key, AttributeValue value) {
+          public <T> void consume(AttributeKey<T> key, T value) {
             tags.add(toKeyValue(key, value));
           }
         });
     return tags;
   }
 
   /**
-   * Converts the given key and {@link AttributeValue} into Jaeger's {@link Model.KeyValue}.
+   * Converts the given {@link AttributeKey} and value into Jaeger's {@link Model.KeyValue}.
    *
    * @param key the entry key as string
    * @param value the entry value
    * @return a Jaeger key value
    */
   @VisibleForTesting
-  static Model.KeyValue toKeyValue(String key, AttributeValue value) {
+  static <T> Model.KeyValue toKeyValue(AttributeKey<T> key, T value) {
     Model.KeyValue.Builder builder = Model.KeyValue.newBuilder();
-    builder.setKey(key);
+    builder.setKey(key.get());
 
-    switch (value.getType()) {
+    switch (key.getType()) {
       case STRING:
-        builder.setVStr(value.getStringValue());
+        builder.setVStr((String) value);
```

Comment:
IIRC, the suggestion for a consumer interface with method per type was to avoid the somewhat lost type-safety of this pattern :) But I don't think it's that bad anyways

Reply:
A (not very pretty) alternative is to have `AttributeKey.getValue(value)` to automatically cast it (and report an error otherwise). Not sure it's a required addition though.

Otherwise, the type safety we lose is not bad.

Reply:
We can also add a secondary consumer interface that has the 8 methods on it. I originally had that when I was working on this, but discovered that it was much more cumbersome to use in basically every case inside the SDK itself. 

@carlosalberto I'm not sure how we could do the casting on the List types, though, since generics are all erased. The best we could do was verify that it was a List, but not what kind, unfortunately.

Reply:
Did you find it cumbersome in the exporters? I'm surprised since I think I see the 8-way switch coming up a few times and figure it'd be similar if separate methods. But might be missing something, definitely agree to optimize the API for the export case since that's almost all of the iterating of attributes in practice.

Reply:
It was super awkward in the exporters that convert to protobuf representations. There might be a better way to approach it, but I wanted to at least reduce the scope of the PR as much as I could. ;) Given that this export is SDK internal functionality, I think we can iterate on it at little more easily than the instrumentation API.